src/test/scala/wdltool/MainSpec.scala

```diff
@@ -0,0 +1,184 @@
+package wdltool
+
+import java.nio.file.{Paths, Path}
+import better.files._
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+import wdltool.SampleWdl.{EmptyWorkflow, EmptyTask, EmptyInvalid, ThreeStep}
+import MainSpec._
+
+class MainSpec extends FlatSpec with Matchers with BeforeAndAfterAll {
+
+  import Main._
+
+  behavior of "Main"
+
+  val threeStep = ThreeStep.wdlSource()
+
+  it should "print usage" in {
+    Main.dispatchCommand(Seq.empty[String]) shouldBe BadUsageTermination
+  }
+
+  it should "validate" in {
+    testWdl(ThreeStep) { wdlAndInputs =>
+      Main.dispatchCommand(Seq("validate", wdlAndInputs.wdl)) shouldBe SuccessfulTermination("")
+    }
+  }
+
+  it should "not validate invalid wdl" in {
+    testWdl(EmptyInvalid) { wdlAndInputs =>
+      val res = Main.dispatchCommand(Seq("validate", wdlAndInputs.wdl))
+      assert(res.isInstanceOf[UnsuccessfulTermination])
+      res.output should include("Finished parsing without consuming all tokens")
+    }
+  }
+
+  it should "parse" in {
+    testWdl(ThreeStep) { wdlAndInputs =>
+      val res = Main.dispatchCommand(Seq("parse", wdlAndInputs.wdl))
+      assert(res.isInstanceOf[SuccessfulTermination])
+      res.output should include("(Document:")
+    }
+  }
+
+  it should "highlight" in {
+    testWdl(ThreeStep) { wdlAndInputs =>
+      val res = Main.dispatchCommand(Seq("highlight", wdlAndInputs.wdl, "html"))
+      assert(res.isInstanceOf[SuccessfulTermination])
+      res.output.stripLineEnd should be(HighlightedWdlHtml)
+    }
+  }
+
+  it should "highlight using console highlighting" in {
+    testWdl(EmptyWorkflow) { wdlAndInputs =>
```

Comment:
I'm not fixing things which were copied straight from cromwell (except references to cromwell!). Feel free to submit your own PR :)